backend/tests/test_auth.py

```
@@ -0,0 +1,108 @@
+from models.user import User
+from core.security import hash_password
+from sqlalchemy.orm import Session
+
+# Initialize FastAPI test client
+client = TestClient(app)
+
+# Ensure tables exist before testing
+Base.metadata.create_all(bind=engine)
+
+# Dependency override to use testing DB session
+def override_get_db():
```

Ruff (0.8.2)

Missing return type annotation for public function `override_get_db`

(ANN201)